packages/richtext-lexical/src/features/toolbars/shared/ToolbarDropdown/index.tsx

```diff
@@ -137,56 +147,59 @@ export const ToolbarDropdown = ({
           _enabledItemKeys.push(item.key)
         }
       }
-      if (group.isEnabled) {
-        setEnabledGroup(group.isEnabled({ editor, editorConfigContext, selection }))
-      }
-      setActiveItemKeys(_activeItemKeys)
-      setEnabledItemKeys(_enabledItemKeys)
+
+      setToolbarState({
+        activeItemKeys: _activeItemKeys,
+        enabledGroup: group.isEnabled
+          ? group.isEnabled({ editor, editorConfigContext, selection })
+          : true,
+        enabledItemKeys: _enabledItemKeys,
+      })
 
       if (onActiveChange) {
         onActiveChange({ activeItems: _activeItems })
       }
     })
   }, [editor, editorConfigContext, group, items, maxActiveItems, onActiveChange])
 
-  useEffect(() => {
-    updateStates()
-  }, [updateStates])
-
   useEffect(() => {
     return mergeRegister(
       editor.registerUpdateListener(() => {
-        updateStates()
+        requestIdleCallback(() => {
+          updateStates()
+        })
```

Comment:
`requestIdleCallback` is not working on Safari. How about using `requestAnimationFrame` like [`await-interaction-response`](https://github.com/vercel-labs/await-interaction-response/blob/main/packages/await-interaction-response/src/interaction-response.ts)?

Reply:
Good idea! I tested it, and it works great on safari